### PR TITLE
Spot

### PR DESCRIPTION
**Pull request by SooLee**
This PR adds a spot instance support for snakemake-tibanna

### CODE REVIEW

File: snakemake/__init__.py [hunk 1]
```diff
@@ -144,6 +144,7 @@ def snakemake(
     precommand="",
     default_remote_provider=None,
     default_remote_prefix="",
+    spot_instance=False,
```
Comment (johanneskoester) [inline snakemake/__init__.py]: I think we should call this `tibanna_spot_instance` and the flag accordingly.

**Comment by johanneskoester** [comment]
And also please format with black. Thank you!

**Comment by SooLee** [comment]
@johanneskoester Thanks for the review! I think we will need to expose more tibanna config parameters, so I'm thinking we could generalize it to something like `--tibanna-configfile` and have a json that defines spot instance, security group, subnet, etc (I had a use case for having to use these parameters). What do you think?

**Comment by johanneskoester** [comment]
> @johanneskoester Thanks for the review! I think we will need to expose more tibanna config parameters, so I'm thinking we could generalize it to something like `--tibanna-configfile` and have a json that defines spot instance, security group, subnet, etc (I had a use case for having to use these parameters). What do you think?

In order to make tibanna straightforwardly configurable via [profiles](https://snakemake.readthedocs.io/en/stable/executing/cli.html#profiles), I'd like to stick with command line flags, all prefixed with --tibanna, and in its own group.

**Comment by johanneskoester** [comment]
Maybe something like `--tibanna-args subnet=xxx spot=true security-group=foo`

**Comment by SooLee** [comment]
@johanneskoester Sounds good! I can use `--tibanna-config` (we use args for a different purpose, so config would be a better term).

**Comment by SooLee** [comment]
@johanneskoester I had to comment out the envvar lines in the Snakefile - apparently the latest Snakemake Docker image doesn't recognize this syntax. It may fail some tests if they're using envvar. 
 I also turned off default resource because the latest image didn't recognize input size_mb.

**Pull request merged**